extensions/api/api-core/src/main/java/org/eclipse/dataspaceconnector/api/query/QuerySpecDto.java

```diff
@@ -0,0 +1,128 @@
+/*
+ *  Copyright (c) 2022 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Bayerische Motoren Werke Aktiengesellschaft (BMW AG) - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.api.query;
+
+import jakarta.validation.constraints.AssertTrue;
+import jakarta.validation.constraints.Positive;
+import jakarta.validation.constraints.PositiveOrZero;
+import jakarta.ws.rs.QueryParam;
+import org.eclipse.dataspaceconnector.spi.query.SortOrder;
+
+public class QuerySpecDto {
+
+    @QueryParam("offset")
+    @PositiveOrZero
+    private Integer offset = 0;
+
+    @QueryParam("limit")
+    @Positive
+    private Integer limit = 50;
+
+    @QueryParam("filter")
+    private String filter;
+
+    @QueryParam("sort")
+    private String sortOrder = "ASC";
+
+    @QueryParam("sortField")
+    private String sortField;
+
+    public QuerySpecDto() {
+
+    }
+
+    public Integer getOffset() {
+        return offset;
+    }
+
+    public Integer getLimit() {
+        return limit;
+    }
+
+    public String getFilter() {
+        return filter;
+    }
+
+    public String getSortOrder() {
+        return sortOrder;
+    }
+
+    public String getSortField() {
+        return sortField;
+    }
+
+    @AssertTrue
```

Comment:
This needs to have the `@JsonIgnore` annotation, otherwise Jackson will look for a `valid` field. A serialization test would have surfaced that. Mind adding one?

Reply:
This class is never serialized to/deserialized from json, so it's not needed

Reply:
I haven't checked, but what about tests (integration, e2e)?

Reply:
test are working just fine without touching them, I just add a "failing query" test for every controller

Reply:
sure, that'll work, but as far as I'm aware @algattik has plans to generate a java API client from our own OpenAPI spec and use that in tests. If that is the case we'd indeed have to serialize the `QuerySpecDto`.

Reply:
@paullatzelsperger I'm not sure about that, however I would leave this for that issue, if it's needed it will be done

Reply:
ok, fine for me.